### PR TITLE
added --min-alt-rna-reads option

### PR DESCRIPTION
**Pull request by iskandr**
CLI args:

`--min-alt-rna-reads`: If a variant has fewer than this number of RNA reads supporting it then skip it before even trying to construct variant cDNA sequences.

`--min-variant-sequence-coverage`: Trim VariantSequence objects to only positions which are spanned by at least this number of variant-supporting reads (and drop the VariantSequence if not bases are sufficiently covered). 

Also, for consistency, I renamed the other function arguments from e.g. `variant_cdna_sequence_assembly` to `variant_sequence_assembly`, since I don't want to maintain two different terms for the same object ("variant sequence" vs. "variant cdna sequence").

**Update:** 

Found a bug in how overlap size between VariantSequences was computed and also saw some room for improvement in the assembly algorithm. Changed both of those and added VariantSequence and assembly unit tests. 

---
This change is [Reviewable](https://reviewable.io/reviews/hammerlab/isovar/67)

### CODE REVIEW

File: isovar/cli/variant_sequences.py [hunk 1]
```diff
@@ -80,9 +88,10 @@ def variant_sequences_generator_from_args(args):
     allele_reads_generator = allele_reads_generator_from_args(args)
     return reads_generator_to_sequences_generator(
         allele_reads_generator,
-        min_reads_supporting_cdna_sequence=args.min_reads_supporting_variant_sequence,
+        min_alt_rna_reads=args.min_alt_rna_reads,
+        min_variant_sequence_coverage=args.min_variant_sequence_coverage,
         preferred_sequence_length=args.cdna_sequence_length,
-        variant_cdna_sequence_assembly=args.variant_sequence_assembly)
+        variant_cdna_sequence_assembly=not args.disable_variant_sequence_assembly)
```
Comment (julia326) [inline isovar/cli/variant_sequences.py]: you renamed this parameter to `variant_sequence_assembly` - think this should break now?

File: isovar/locus_reads.py [hunk 2]
```diff
@@ -107,6 +107,7 @@ def from_pysam_pileup_element(
         mapping_quality = read.mapping_quality
 
         missing_mapping_quality = mapping_quality is None
+
```
Comment (julia326) [inline isovar/locus_reads.py]: random newline: revert and you can leave this file out of the PR

File: isovar/cli/variant_sequences.py (hunk 1 above)
Comment (iskandr) [inline isovar/cli/variant_sequences.py]: Yep, good catch

File: isovar/cli/variant_sequences.py [hunk 3]
```diff
@@ -31,24 +32,31 @@
...
         type=int,
-        default=MIN_READS_SUPPORTING_VARIANT_CDNA_SEQUENCE,
+        default=MIN_VARIANT_SEQUENCE_COVERAGE,
         help="Minimum number of reads supporting a variant sequence (default %(default)s)")
 
     rna_sequence_group.add_argument(
-        "--variant-sequence-assembly",
-        default=False,
-        action="store_true")
+        "--disable-variant-sequence-assembly",
```
Comment (julia326) [inline isovar/cli/variant_sequences.py]: why not default to `--variant-sequence-assembly True` in that case? then the user can disable it if they need to

File: test/test_assembly.py [hunk 4]
```diff
@@ -15,10 +15,13 @@
 from __future__ import print_function, division, absolute_import
 
 from isovar.variant_reads import reads_supporting_variant
-from isovar.variant_sequences import initial_variant_sequences_from_reads
+from isovar.variant_sequences import (
+    initial_variant_sequences_from_reads,
+    VariantSequence
```
Comment (julia326) [inline test/test_assembly.py]: why import VariantSequence here? Doesn't seem like you use it in any of the other changes to this file. Also, seems unrelated to this PR - ideally this and other unrelated test file changes should be in a different branch/PR - okay for now but try not to add any more here :)

File: isovar/cli/variant_sequences.py (hunk 3 above)
Comment (iskandr) [inline isovar/cli/variant_sequences.py]: I'm confused about what you mean. Explain more?

File: test/test_assembly.py (hunk 4 above)
Comment (iskandr) [inline test/test_assembly.py]: I combined two different import lines into one. (see the deleted line below). 

I can try to keep these kinds of changes out of PRs but it's also the kind of small cleanup that would never happen otherwise (since it's hard to remember to do it in some other PR).

File: isovar/cli/variant_sequences.py (hunk 3 above)
Comment (julia326) [inline isovar/cli/variant_sequences.py]: Ah - yes, much prefer to disable via `--variant-sequence-assembly false`.

File: test/test_assembly.py (hunk 4 above)
Comment (julia326) [inline test/test_assembly.py]: my bad, didn't notice that it got moved, thought it just got added

File: isovar/cli/translation.py [hunk 5]
```diff
@@ -78,7 +78,9 @@ def translations_generator_from_args(args):
     return translate_variants(
         variant_reads_generator,
         protein_sequence_length=args.protein_sequence_length,
-        min_reads_supporting_cdna_sequence=args.min_reads_supporting_variant_sequence,
+        min_alt_rna_reads=args.min_alt_rna_reads,
+        min_variant_sequence_coverage=args.min_variant_sequence_coverage,
+        variant_sequence_assembly=not args.disable_variant_sequence_assembly,
```
Comment (julia326) [inline isovar/cli/translation.py]: at this point you can change this to `variant_sequence_assembly=args.variant_sequence_assembly` right?

Comment (iskandr) [inline isovar/cli/translation.py]: 👍

**Comment by iskandr** [comment]
Closing in favor of https://github.com/hammerlab/isovar/pull/68
